ops/_main.py

```diff
@@ -217,43 +151,24 @@ def __init__(self, charm_dir: Path, juju_context: _JujuContext):
         self._charm_dir = charm_dir
         self._exec_path = Path(self._juju_context.dispatch_path or sys.argv[0])
 
-        dispatch = charm_dir / 'dispatch'
-        if self._juju_context.version.is_dispatch_aware() and _exe_path(dispatch) is not None:
-            self._init_dispatch()
-        else:
-            self._init_legacy()
+        # Grab the wanted hook from JUJU_DISPATCH_PATH, e.g. hooks/install.
```

Comment:
Nit: "wanted hook" reads a bit funny here. How about:
```suggestion
        # Grab the correct hook from JUJU_DISPATCH_PATH, e.g. hooks/install.
```